### PR TITLE
`violin`: handle zero standard deviation

### PR DESCRIPTION
**Pull request by antoinerg**
Closes issue #3622 

This is by far the most concise way I could fix the issue. I only added 4 lines to `src/traces/violin/calc.js` (starting from line 38).

### CODE REVIEW

File: src/traces/violin/calc.js [hunk 1]
```diff
@@ -33,28 +33,36 @@ module.exports = function calc(gd, trace) {
...
-        for(var k = 0, t = span[0]; t < (span[1] + step / 2); k++, t += step) {
-            var v = kde(t);
-            cdi.density[k] = {v: v, t: t};
-            maxKDE = Math.max(maxKDE, v);
+        if (cdi.min === cdi.max) {
+            cdi.spanZero = true;
+            span = cdi.span = [cdi.min, cdi.max];
+            cdi.density = [{v:1, t: span[0]}];
+            maxKDE = Math.max(maxKDE, 1);
+        } else {
```
Comment (antoinerg) [inline src/traces/violin/calc.js]: The following is exactly the same code we had before.

**Comment by alexcjohnson** [comment]
Much better 🍻
Question though: what if someone *wants* a nonzero bandwidth despite the single-valued data? Ideally they should still be able to provide a `trace.bandwidth` and get that behavior. I might even suggest doing that in the `groups-over-matching-axes` and `violin-offsetgroups` mocks to restore the previous look and make it easier to see what's going on.

File: src/traces/violin/calc.js [hunk 2]
```diff
@@ -33,28 +33,37 @@ module.exports = function calc(gd, trace) {
...
-        cdi.density = new Array(n);
-
-        for(var k = 0, t = span[0]; t < (span[1] + step / 2); k++, t += step) {
-            var v = kde(t);
-            cdi.density[k] = {v: v, t: t};
-            maxKDE = Math.max(maxKDE, v);
+        var span;
+
+        if(cdi.min === cdi.max) {
+            cdi.spanZero = true;
```
Comment (alexcjohnson) [inline src/traces/violin/calc.js]: Is this used?

**Comment by antoinerg** [comment]
> Question though: what if someone wants a nonzero bandwidth despite the single-valued data? Ideally they should still be able to provide a trace.bandwidth and get that behavior. I might even suggest doing that in the groups-over-matching-axes and violin-offsetgroups mocks to restore the previous look and make it easier to see what's going on.

Right... I was approaching this the wrong way. Closing in favor of https://github.com/plotly/plotly.js/pull/3626

**Pull request closed without merging**